ipu-plugin/pkg/ipuplugin/bridgeport.go

```diff
@@ -83,48 +87,115 @@ func isBridgePortPresent(srv server, brPortName string) bool {
+		}
+		log.Debugf("outer vlan interface %s already exist", vlanIntfName)
+		return nil
+	}
+
+	vlan := netlink.Vlan{}
+	vlan.ParentIndex = upLink.Attrs().Index
+	vlan.Name = vlanIntfName
+	vlan.VlanId = vlanId
+
+	log.Debugf("creating a new outer vlan interface %s", vlanIntfName)
+	if err := linkAddFn(&vlan); err != nil {
+		return fmt.Errorf("error creating outer vlan interface %s: %s", vlanIntfName, err)
+	}
+	log.Debugf("outer vlan interface %s is created", vlanIntfName)
+
+	return nil
+}
+
+func createAndSetUpInnerVlan(uplinkInterface string, bridges []string) (string, error) {
+	// Assume that the uplink interface name is something like enp0s1f0d3
+	// take only the last two characters from the name to avoid long names limit
+	outerVlanIntfName := fmt.Sprintf("%v.%d", uplinkInterface[len(uplinkInterface)-2:], outerVlanId)
+
+	upLink, err := linkByNameFn(outerVlanIntfName)
+	if err != nil {
+		if err == err.(netlink.LinkNotFoundError) {
+
+			// Outer vlan interface has not been created yet
+			// We create it here
+			_, err := createAndSetUpOuterVlan(uplinkInterface)
+			if err != nil {
+				return "", fmt.Errorf("unable to create outer vlan interface %s, because: %w", outerVlanIntfName, err)
+			}
+
+			upLink, err = linkByNameFn(outerVlanIntfName)
+
+			if err != nil {
+				return "", fmt.Errorf("unable to find uplink interface: %s, because: %w", uplinkInterface, err)
+			}
+
+		} else {
+			return "", fmt.Errorf("unable to find uplink interface: %s, because: %w", uplinkInterface, err)
+		}
+	}
+
+	innerVlanId, err := strconv.Atoi(bridges[0])
 	if err != nil {
 		return "", fmt.Errorf("unable to parse vlan ID: %s, because: %w", bridges[0], err)
 	}
 
-	vlanIntfName := fmt.Sprintf("%s_vlan%d", vlanName, vlanId)
-	if err := createVlanInterface(upLink, vlanIntfName, vlanId); err != nil {
+	// Assume that the uplink interface name is something like enp0s1f0d3
+	// take only the last two characters from the name to avoid long names limit
+	vlanIntfName := fmt.Sprintf("%v.%d.%d", uplinkInterface[len(uplinkInterface)-2:], outerVlanId, innerVlanId)
+
+	if err := createInnerVlanInterface(upLink, vlanIntfName, outerVlanId); err != nil {
 		return "", err
 	}
 
 	return vlanIntfName, nil
 }
 
-func createVlanInterface(upLink netlink.Link, vlanIntfName string, vlanId int) error {
+func createInnerVlanInterface(upLink netlink.Link, vlanIntfName string, vlanId int) error {
 
 	link, err := linkByNameFn(vlanIntfName)
 	if err == nil {
 		// If the vlan interface already exist then do nothing and return
 		if _, ok := link.(*netlink.Vlan); !ok {
 			return fmt.Errorf("an interface %s is found but not a VLAN device", vlanIntfName)
 		}
-		log.Debugf("vlan interface %s already exist", vlanIntfName)
+		log.Debugf("inner vlan interface %s already exist", vlanIntfName)
 		return nil
 	}
 
 	vlan := netlink.Vlan{}
 	vlan.ParentIndex = upLink.Attrs().Index
 	vlan.Name = vlanIntfName
 	vlan.VlanId = vlanId
+	vlan.VlanProtocol = netlink.VLAN_PROTOCOL_8021AD
```

Comment:
I think you meant to add `VlanProtocol` on outerVlan interface.

Reply:
Addressed